storage/innobase/handler/ha_innodb.cc

```diff
@@ -2580,6 +2580,7 @@ ha_innobase::ha_innobase(
                           | HA_CAN_ONLINE_BACKUPS
 			  | HA_CONCURRENT_OPTIMIZE
 			  |  (srv_force_primary_key ? HA_REQUIRE_PRIMARY_KEY : 0)
+			  | HA_DUPLICATE_KEY_NOT_IN_ORDER
```

Comment:
Instead of setting this flag for all InnoDB table handles, could we execute `m_int_table_flags|= HA_DUPLICATE_KEY_NOT_IN_ORDER` in `check_index_consistency()` only when the InnoDB unique indexes are defined in a different order from the `TABLE_SHARE`? It seems that we should also rename that function to `ha_innobase::check_index_consistency(const dict_table_t&) noexcept`.